### PR TITLE
docs: Improve trace_dns guide

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
Improve trace_dns guide. 

### Testing done:

Rendered using: https://github.com/inspektor-gadget/website

image

### CODE REVIEW

File: gadgets/trace_dns/README.mdx [hunk 1]
```diff
@@ -19,52 +19,165 @@ Running the gadget:
         ```bash
         $ kubectl gadget run ghcr.io/inspektor-gadget/gadget/trace_dns:latest [flags]
         ```
+        :::tip
+        If you want to try the gadget without installation you can check [kubectl node debug](https://inspektor-gadget.io/docs/latest/quick-start#kubectl-node-debug).
```
Comment (alban) [inline gadgets/trace_dns/README.mdx:23]: Should we add those tips in all gadgets?

File: gadgets/trace_dns/README.mdx [hunk 2]
```diff
@@ -19,52 +19,165 @@ Running the gadget:
...
+        ```sh
+        nslookup -query=a inspektor-gadget.io.
+        ```
+
+        The request/response will be logged by the DNS gadget:
+
+        ```
+        K8S.NODE             K8S.NAMESPACE        K8S.PODNAME          K8S.CONTAINERNAME    SRC                       DST                       COMM              PID QR QTYPE      NAME                RCODE    ADDRESSES
+        minikube-docker      demo                 mypod                mypod                10.244.0.4:56093          10.96.0.10:53             isc-net-0…     422204 Q  A          inspektor-gadget.i…
+        minikube-docker      demo                 mypod                mypod                10.96.0.10:53             10.244.0.4:56093          isc-net-0…     422204 R  A          inspektor-gadget.i… Success  172.67.16…
```
Comment (alban) [inline gadgets/trace_dns/README.mdx:112]: Could we have an output where the name and addresses are not truncated?

Why are src/dst not enriched?

**Review by alban — COMMENTED** [review]
This looks good.

Also, this guide could easily be used for a killercoda tutorial: https://github.com/inspektor-gadget/learn-ig

File: gadgets/trace_dns/README.mdx (hunk 1 above)
Comment (mqasimsarfraz) [reply gadgets/trace_dns/README.mdx]: Good point, I moved it to: https://github.com/inspektor-gadget/inspektor-gadget/pull/3433

**Comment by mqasimsarfraz** [comment]
> Also, this guide could easily be used for a killercoda tutorial (https://github.com/inspektor-gadget/learn-ig).
> 
> There are also scenario using coredns already (https://killercoda.com/explore?search=coredns&type=scenario). Maybe some of them can be improved with using Inspektor Gadget.

Great, thanks for sharing. I will have a look!

File: gadgets/trace_dns/README.mdx (hunk 2 above)
Comment (mqasimsarfraz) [reply gadgets/trace_dns/README.mdx]: IIUC, [kubeipresolver](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/operators/kubemanager/kubemanager.go) hasn't been adopted to work for image-based gadgets unless am I missing something. @flyth or @mauriciovasquezbernal do you have any thoughts?

Comment (mauriciovasquezbernal) [reply gadgets/trace_dns/README.mdx]: I documented it by mistake on #3406 but it doesn't work with image-based gadgets yet.

File: gadgets/trace_dns/README.mdx [hunk 3]
```diff
@@ -26,45 +26,152 @@ Running the gadget:
...
+        minikube-docker                                                              7f22          OUTGOING       4026532708  10.96.0.10:53                                        10.244.0.4:51340         R  inspektor-gadget.io.       Success    0          2024-08-30T15:43:35.493416278Z
+        minikube-docker         demo                 mypod                           7f22          HOST           4026533235  10.96.0.10:53                                        10.244.0.4:51340         R  inspektor-gadget.io.       Success    30245219   2024-08-30T15:43:35.493418950Z
+        ```
+
+        The output is same as previous scenario, but now we provides additional context related to what is happening to DNS traffic at node level. For example, the second/third line show how packet is forwarded via the host (`NETNS_ID=4026532708`) and how kube-dns service IP is translated (e.g. by `iptables`) to specific pod IP. Similarly you can use other events with `NETNS_ID=4026532708` (or without kubernetes enrichment) to understand how the packet is handled at host level.
```
Comment (mauriciovasquezbernal) [inline gadgets/trace_dns/README.mdx:172]: we provides -> we provide OR it provides?

**Review by mauriciovasquezbernal — APPROVED** [review]
Just a minor nit. LGTM otherwise, great job!

**Comment by mqasimsarfraz** [comment]
Thanks for the reviews!

**Pull request merged**